arches_lingo/src/arches_lingo/components/concept/ConceptImages/components/ConceptImagesViewer.vue

```diff
@@ -0,0 +1,195 @@
+<script setup lang="ts">
+import { inject, ref, onMounted } from "vue";
+import { useGettext } from "vue3-gettext";
+
+import Button from "primevue/button";
+
+import FileListWidget from "@/arches_component_lab/widgets/FileListWidget/FileListWidget.vue";
+
+import { DANGER, SECONDARY, VIEW } from "@/arches_lingo/constants.ts";
+import { useConfirm } from "primevue/useconfirm";
+
+import type {
+    ConceptImages,
+    DigitalObjectInstance,
+} from "@/arches_lingo/types.ts";
+import { fetchLingoResource } from "@/arches_lingo/api.ts";
+import NonLocalizedStringWidget from "@/arches_component_lab/widgets/NonLocalizedStringWidget/NonLocalizedStringWidget.vue";
+
+const props = defineProps<{
+    tileData: ConceptImages | undefined;
+    componentName: string;
+    sectionTitle: string;
+    graphSlug: string;
+    nodegroupAlias: string;
+}>();
+
+const resources = ref<DigitalObjectInstance[]>();
+const { $gettext } = useGettext();
+const confirm = useConfirm();
+
+onMounted(async () => {
+    if (props.tileData) {
+        resources.value = await Promise.all(
+            props.tileData.aliased_data.depicting_digital_asset_internal.map(
+                async (digitalAsset) =>
+                    await getConceptImageResource(digitalAsset.resourceId),
+            ),
+        );
+    }
+});
+function confirmDelete() {
+    confirm.require({
+        header: $gettext("Confirmation"),
+        message: $gettext("Do you want to delete this concept image?"),
+        accept: () => {
+            console.log("do delete");
+        },
+        rejectProps: {
+            label: $gettext("Cancel"),
+            severity: SECONDARY,
+            outlined: true,
+        },
+        acceptProps: {
+            label: $gettext("Delete"),
+            severity: DANGER,
+        },
+    });
+}
+
+const openEditor = inject<(componentName: string) => void>("openEditor");
+async function getConceptImageResource(resourceInstanceId: string) {
```

Comment:
will the component need a loading state to handle this?

Reply:
Yep good call